File: arrayPairSum.js
function pair_sum(arr,k) {
    // Edge Case
    if(arr.length < 2) {
        return 
    }

    var seen = new Set();
    var output = [];

    // Iterate over the arr using map
    arr.map(function(num) {
        const target = k - num;
    // if the target exist in seen object, we have found the compliment value    
        if(seen.has(target)) {
            output.push([target,num])
        } else {
            seen.add(num)
        }
    })
    return output
}

console.log(pair_sum([1, 3, 2, 2], 4));

File: arrayPairSum.py
def pair_sum(arr, k):
    # Edge Case
    if len(arr) < 2:
        return

    # sets for tracking
    seen = set()
    output = set()

    for num in arr:
        target = k - num

        if target not in seen:
            seen.add(num)
        else:
            output.add( ((min(num,target)), max(num,target)) )

    # return len(output) 
    print '\n'.join(map(str,list(output)))       

pair_sum([1, 3, 1, 4], 5)

